parquet/src/main/java/org/apache/iceberg/data/parquet/BaseParquetWriter.java

```diff
@@ -50,6 +54,32 @@ protected ParquetValueWriter<T> createWriter(MessageType type) {
   protected abstract ParquetValueWriters.StructWriter<T> createStructWriter(
       List<ParquetValueWriter<?>> writers);
 
+  protected ParquetValueWriter<?> fixedWriter(ColumnDescriptor desc) {
+    return new FixedWriter(desc);
+  }
+
+  protected ParquetValueWriters.PrimitiveWriter<?> dateWriter(ColumnDescriptor desc) {
```

Comment:
These methods should return `ParquetValueWriter<?>` and not a specific class.

Reply:
Existing class `LogicalTypeWriterVisitor` have declared return values as `Optional<ParquetValueWriters.PrimitiveWriter<?>>`. So, I need to modify the base class also to return `ParquetValueWriter<?>`. 

I thought I will get a comment that why we modified base class and avoid unnecessary refactoring!
Hence, I didn't do. 

Since, you also want to handle this. I will update it.

Reply:
Returning the right type (in this case, the interface) is a good thing. It's worth refactoring the private visitor.